Taller 2 Curso: Estadística y Análisis Exploratorio

* Docente: Javier Lozano
* Estudiante: Juan Pablo Mazo Quintero y Danilo Rodriguez Arango 
* Tema: Estimaciones 

La base de datos con la que le corresponde trabajar, se obtiene como una muestra
aleatoria que corresponde a corresponde a los resultados obtenidos por los jóvenes de
todo el país en las pruebas Saber 11, del año 2022. La información aparece agregada
por Colegio y solo abarca los colegios de Antioquia. Dicha base contiene las variables:
NATURALEZA (Colegio Oficial o No-Oficial), JORNADA (Asociado a los diferentes
tipos de jornada de estudio), PROMLECT (Promedio obtenido por el colegio en Lectura
Crítica) y PROMMATE (Promedio obtenido por el colegio en Matemáticas).




1. Construir en Python un IC para al 95 % de confianza necesario para responder:
¿Puede afirmarse que el resultado medio obtenido por los colegios en Lectura es
superior a 65 puntos? No asuma normalidad, justifique su respuesta.

In [1]:
import pandas as pd
import numpy as np
import scipy.stats as stats

In [7]:
data = pd.read_csv('DatosTrabajo2EAE20242.txt')
data.info()

<class 'pandas.core.frame.DataFrame'>
RangeIndex: 1807 entries, 0 to 1806
Data columns (total 4 columns):
 #   Column      Non-Null Count  Dtype 
---  ------      --------------  ----- 
 0   NATURALEZA  1807 non-null   object
 1   JORNADA     1807 non-null   object
 2   PROMLECT    1807 non-null   int64 
 3   PROMMATE    1807 non-null   int64 
dtypes: int64(2), object(2)
memory usage: 56.6+ KB


In [23]:
data.describe()

PROMLECT     PROMMATE
count  1807.000000  1807.000000
mean     50.321527    47.105147
std       7.028032     7.897110
min       0.000000    18.000000
25%      45.000000    41.000000
50%      50.000000    47.000000
75%      55.000000    52.000000
max      74.000000    76.000000

In [22]:
# Parámetros 
n = len(data)
mean_lectura = data['PROMLECT'].mean()
std_dev_lectura = data['PROMLECT'].std(ddof=1)

# Valor t crítico para un intervalo de confianza del 95%
t_critical = stats.t.ppf(1 - 0.025, n -1)

# Cálculo del margen de error
margin_of_error = t_critical * (std_dev_lectura / np.sqrt(n))

# Calcular el intervalo de confianza
lower_bound = mean_lectura - margin_of_error
upper_bound = mean_lectura + margin_of_error
print(f"Intervalo de confianza del 95% para PROMLECT: ({lower_bound:.2f}, {upper_bound:.2f})")

Intervalo de confianza del 95% para PROMLECT: (50.00, 50.65)


Interpretación: Este intervalo de confianza del 95% indica que, si realizáramos muchas muestras aleatorias de colegios en Antioquia y calculáramos el promedio de Lectura para cada una, el 95% de esos promedios de muestra caerían dentro del rango de 50.00 a 50.65 puntos.

2. ¿Se puede afirmar que el resultado medio obtenido por los colegios en Matemáticas
es mayor en los Privados que en los Públicos? Construir en Python un IC para al
96 % de confianza sin asumir normalidad, justifique su respuesta.

In [24]:
# Parámetros para colegios públicos y privados
n_privado = len(data[data['NATURALEZA'] == 'NO OFICIAL'])
mean_privado = data[data['NATURALEZA'] == 'NO OFICIAL']['PROMMATE'].mean()
std_dev_privado = data[data['NATURALEZA'] == 'NO OFICIAL']['PROMMATE'].std(ddof=1)

n_publico = len(data[data['NATURALEZA'] == 'OFICIAL'])
mean_publico = data[data['NATURALEZA'] == 'OFICIAL']['PROMMATE'].mean()
std_dev_publico = data[data['NATURALEZA'] == 'OFICIAL']['PROMMATE'].std(ddof=1)

# Grados de libertad
df = ((std_dev_privado**2 / n_privado + std_dev_publico**2 / n_publico)**2) / \
     (((std_dev_privado**2 / n_privado)**2 / (n_privado - 1)) + ((std_dev_publico**2 / n_publico)**2 / (n_publico - 1)))

# Valor t crítico para un intervalo de confianza del 96%
t_critical = stats.t.ppf(1 - 0.02, df)

# Cálculo del margen de error para la diferencia de medias
margin_of_error = t_critical * np.sqrt((std_dev_privado**2 / n_privado) + (std_dev_publico**2 / n_publico))

# Diferencia de medias
difference_of_means = mean_privado - mean_publico

# Intervalo de confianza
lower_bound = difference_of_means - margin_of_error
upper_bound = difference_of_means + margin_of_error

# Mostrar resultados
print(f"El intervalo de confianza del 96% para la diferencia de medias entre los colegios privados y públicos es: [{lower_bound:.2f}, {upper_bound:.2f}]")

El intervalo de confianza del 96% para la diferencia de medias entre los colegios privados y públicos es: [5.83, 8.09]


Interpretación: Con un intervalo de confianza del 96% podemos determinar que hay una diferencia en los promedios de Matemáticas entre los colegios privados y públicos se encuentra entre 5.83 y 8.09 puntos